### PR TITLE
develop.bio labels

### PR DESCRIPTION
**Pull request by sinistral**
Attach labels to the BIO entities for proteins and genes.  There are a couple of points worth noting:

* no special effort is made to ensure that one, and only one, label is attached to the entity; it is unclear whether this is actually desirable in the long term (we may, for example, decide to use a different property to describe the "preferred" name).
* The labels are simply the ID, without any indication of its origin; this should be changed in (the near) future so that a user/developer won't have to guess at the original source of the label.

To address either of these points, further changes would be needed to the way that the rules are interpreted.  We'd need for the rules engine to handle more SPARQL operators (i.e the changes that Elizabeth has made for the Reactome rules), a pure-SPARQL rule body, or more flexible `:head` management.  For example, to attach one and only one rule, one option would be to `MINUS` triples that already have a label; decorating the label with the source name or description would require the ability to `BIND` in body pattern, which is not currently possible.

This rules should, however, satisfy the requirements for the VEGF demo, and so I think they're a useful interim solution.  Counter-{arguments, proposals} are welcome.

### CODE REVIEW

File: kabob-build/src/main/resources/edu/ucdenver/ccp/kabob/build/rules/bio_labels/proteins.clj [hunk 1]
```diff
@@ -19,21 +19,58 @@
...
+ '((?/bio rdfs/label ?/proteinName))
+
+ ;; In this case no `:reify` entry is required, since all of the data required
+ ;; to form the new triples already exist.
+
+ :options
+ {:magic-prefixes
+  [;; AllegroGraph processing directives; http://franz.com/agraph/support/documentation/current/sparql-reference.html#header2-138
+
+   ;; Note: DO NOT set the clause reorderer to `identity`.  The default
```
Comment (bill-baumgartner) [inline kabob-build/src/main/resources/edu/ucdenver/ccp/kabob/build/rules/bio_labels/proteins.clj]: If I had to guess, I would say that swapping lines 16 and 17 in the swiss-prot rule and 64 and 65 in the trembl rule would remedy the slow query times you are observing when using the `identity` clause reorderer. As it stands, lines 16 and 64 are completely disconnected from the beginning parts of the queries (i.e. the line contains no bound variables) and probably result in the collection of all possible records, which are then filtered in the next line based on the bound _/accessionRecordField.

File: kabob-build/src/main/resources/edu/ucdenver/ccp/kabob/build/rules/bio_labels/genes.clj [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+ ;; The query pattern used to retrieve the data that will be used to construct
+ ;; the new triples.
+ :body
+ '(;; field value that contains the identifier
+   (_/gene_id_field kiao/hasTemplate iaoeg/EntrezGeneInfoFileData_geneIDDataField1)
+   (_/gene_id_field rdf/type kiao/FieldValue)
+   ;; the file record in which the field value appeared (record has-part
+   ;; field-value)
+   (_/record rdf/type iaoeg/EntrezGeneInfoFileData)
```
Comment (bill-baumgartner) [inline kabob-build/src/main/resources/edu/ucdenver/ccp/kabob/build/rules/bio_labels/genes.clj]: Similar to the UniProt label rules, there is no bound variable on line 12 but there is on line 13, so swapping lines 12 and 13 may increase performance when using the `identity` clause reorderer, i.e when not permitting AG to reorder the clauses.

**Comment by bill-baumgartner** [comment]
The comments in these rules are terrific. Really made it easy to understand what they are doing.

File: kabob-build/src/main/resources/edu/ucdenver/ccp/kabob/build/rules/bio_labels/proteins.clj (hunk 1 above)
Comment (sinistral) [inline kabob-build/src/main/resources/edu/ucdenver/ccp/kabob/build/rules/bio_labels/proteins.clj]: You're absolutely right.  The order makes a big difference if we were to use the `identity` clause reorderer.  But, assuming I'm interpreting the query plan numbers correctly, the `statistical` clause reorderer appears to do a much better job than manually ordering the clauses.  The `cursor-estimate` costs are the same, but the `triple-query`cost is much lower for the `statistical` reorderer.

**Pull request merged**